.githooks/pre-commit

```diff
@@ -14,10 +14,16 @@ fi
 exec 1>&2
 
 # Check changed files for an AWS keys
-KEY_ID=$(git diff --cached --name-only -z $against | xargs -0 cat | perl -nle'print $& if m{(?<![A-Z0-9])[A-Z0-9]{20}(?![A-Z0-9])}')
-KEY=$(git diff --cached --name-only -z $against | xargs -0 cat | perl -nle'print $& if m{(?<![A-Z0-9])[A-Z0-9]{20}(?![A-Z0-9])}')
-
-if [ "$KEY_ID" != "" -o "$KEY" != "" ]; then
+KEY=$(
+    git diff --cached --name-only -z HEAD~2 |
+        tr '\0' '\n' |
+        grep --invert-match --fixed-strings 'yarn.lock' |
+        xargs ls -d 2> /dev/null |
```

Comment:
For some reason xargs when ran with null characters as separators ignores missing files, but it doesn't when files are piped in individually, so this line filters out missing files

This is the only line i copied off stack overflow 👀